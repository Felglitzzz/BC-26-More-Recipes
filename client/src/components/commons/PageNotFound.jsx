import React from 'react';
import Footer from './Footer';
import zero from '../../../images/zero.svg';

/**
 * @description - Stateless component for rendering a 404 page
 *
 * @returns {view} PageNotFound - Rendered view
 */
function PageNotFound() {
  return (
    <div className="body">
      <header id="not-found">
        <center>
          <a href="/" className="brand-logo">
            More-Recipes
          </a>
        </center>
      </header >
      <main>
        <div className="push-down">
          <div className="not-found">
            <h1>4</h1>
            <img src={zero} alt="" />
            <h1>4</h1>
          </div>
          <center>
            <h2>Oh no...the page you requested was not found</h2>
            <a className="ui brown large button" href="/">
              &gt;&gt; Home &lt;&lt;
            </a>
          </center>
        </div>
      </main>
      <Footer />
    </div >
  );
}

export default PageNotFound;
