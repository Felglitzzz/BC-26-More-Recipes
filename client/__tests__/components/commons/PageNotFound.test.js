import React from 'react';
import { shallow } from 'enzyme';
import PageNotFound from './../../../../client/src/components/commons/PageNotFound';

describe('Component: PageNotFound', () => {
  it('renders properly', () => {
    const wrapper = shallow(<PageNotFound />);
    expect(wrapper).toMatchSnapshot();
  });
});
