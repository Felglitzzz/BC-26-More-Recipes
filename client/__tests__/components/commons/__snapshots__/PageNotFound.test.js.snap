// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PageNotFound renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageNotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          id="not-found"
        >
          <center>
            <a
              className="brand-logo"
              href="/"
            >
              More-Recipes
            </a>
          </center>
        </header>,
        <main>
          <div
            className="push-down"
          >
            <div
              className="not-found"
            >
              <h1>
                4
              </h1>
              <img
                alt=""
                src="efo"
              />
              <h1>
                4
              </h1>
            </div>
            <center>
              <h2>
                Oh no...the page you requested was not found
              </h2>
              <a
                className="ui brown large button"
                href="/"
              >
                &gt;&gt; Home &lt;&lt;
              </a>
            </center>
          </div>
        </main>,
        <FooterView />,
      ],
      "className": "body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <center>
            <a
              className="brand-logo"
              href="/"
            >
              More-Recipes
            </a>
          </center>,
          "id": "not-found",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="brand-logo"
              href="/"
            >
              More-Recipes
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "More-Recipes",
              "className": "brand-logo",
              "href": "/",
            },
            "ref": null,
            "rendered": "More-Recipes",
            "type": "a",
          },
          "type": "center",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="push-down"
          >
            <div
              className="not-found"
            >
              <h1>
                4
              </h1>
              <img
                alt=""
                src="efo"
              />
              <h1>
                4
              </h1>
            </div>
            <center>
              <h2>
                Oh no...the page you requested was not found
              </h2>
              <a
                className="ui brown large button"
                href="/"
              >
                &gt;&gt; Home &lt;&lt;
              </a>
            </center>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="not-found"
              >
                <h1>
                  4
                </h1>
                <img
                  alt=""
                  src="efo"
                />
                <h1>
                  4
                </h1>
              </div>,
              <center>
                <h2>
                  Oh no...the page you requested was not found
                </h2>
                <a
                  className="ui brown large button"
                  href="/"
                >
                  &gt;&gt; Home &lt;&lt;
                </a>
              </center>,
            ],
            "className": "push-down",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    4
                  </h1>,
                  <img
                    alt=""
                    src="efo"
                  />,
                  <h1>
                    4
                  </h1>,
                ],
                "className": "not-found",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "4",
                  },
                  "ref": null,
                  "rendered": "4",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "efo",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "4",
                  },
                  "ref": null,
                  "rendered": "4",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Oh no...the page you requested was not found
                  </h2>,
                  <a
                    className="ui brown large button"
                    href="/"
                  >
                    &gt;&gt; Home &lt;&lt;
                  </a>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Oh no...the page you requested was not found",
                  },
                  "ref": null,
                  "rendered": "Oh no...the page you requested was not found",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": ">> Home <<",
                    "className": "ui brown large button",
                    "href": "/",
                  },
                  "ref": null,
                  "rendered": ">> Home <<",
                  "type": "a",
                },
              ],
              "type": "center",
            },
          ],
          "type": "div",
        },
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            id="not-found"
          >
            <center>
              <a
                className="brand-logo"
                href="/"
              >
                More-Recipes
              </a>
            </center>
          </header>,
          <main>
            <div
              className="push-down"
            >
              <div
                className="not-found"
              >
                <h1>
                  4
                </h1>
                <img
                  alt=""
                  src="efo"
                />
                <h1>
                  4
                </h1>
              </div>
              <center>
                <h2>
                  Oh no...the page you requested was not found
                </h2>
                <a
                  className="ui brown large button"
                  href="/"
                >
                  &gt;&gt; Home &lt;&lt;
                </a>
              </center>
            </div>
          </main>,
          <FooterView />,
        ],
        "className": "body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <center>
              <a
                className="brand-logo"
                href="/"
              >
                More-Recipes
              </a>
            </center>,
            "id": "not-found",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="brand-logo"
                href="/"
              >
                More-Recipes
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "More-Recipes",
                "className": "brand-logo",
                "href": "/",
              },
              "ref": null,
              "rendered": "More-Recipes",
              "type": "a",
            },
            "type": "center",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="push-down"
            >
              <div
                className="not-found"
              >
                <h1>
                  4
                </h1>
                <img
                  alt=""
                  src="efo"
                />
                <h1>
                  4
                </h1>
              </div>
              <center>
                <h2>
                  Oh no...the page you requested was not found
                </h2>
                <a
                  className="ui brown large button"
                  href="/"
                >
                  &gt;&gt; Home &lt;&lt;
                </a>
              </center>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="not-found"
                >
                  <h1>
                    4
                  </h1>
                  <img
                    alt=""
                    src="efo"
                  />
                  <h1>
                    4
                  </h1>
                </div>,
                <center>
                  <h2>
                    Oh no...the page you requested was not found
                  </h2>
                  <a
                    className="ui brown large button"
                    href="/"
                  >
                    &gt;&gt; Home &lt;&lt;
                  </a>
                </center>,
              ],
              "className": "push-down",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1>
                      4
                    </h1>,
                    <img
                      alt=""
                      src="efo"
                    />,
                    <h1>
                      4
                    </h1>,
                  ],
                  "className": "not-found",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "4",
                    },
                    "ref": null,
                    "rendered": "4",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "src": "efo",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "4",
                    },
                    "ref": null,
                    "rendered": "4",
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h2>
                      Oh no...the page you requested was not found
                    </h2>,
                    <a
                      className="ui brown large button"
                      href="/"
                    >
                      &gt;&gt; Home &lt;&lt;
                    </a>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Oh no...the page you requested was not found",
                    },
                    "ref": null,
                    "rendered": "Oh no...the page you requested was not found",
                    "type": "h2",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": ">> Home <<",
                      "className": "ui brown large button",
                      "href": "/",
                    },
                    "ref": null,
                    "rendered": ">> Home <<",
                    "type": "a",
                  },
                ],
                "type": "center",
              },
            ],
            "type": "div",
          },
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
